moabb/evaluations/splitters.py

```diff
@@ -0,0 +1,76 @@
+from sklearn.model_selection import BaseCrossValidator, StratifiedKFold
+from sklearn.utils import check_random_state
+
+
+class WithinSessionSplitter(BaseCrossValidator):
+    """Data splitter for within session evaluation.
+
+    Within-session evaluation uses k-fold cross_validation to determine train
+    and test sets for each subject in each session. This splitter
+    assumes that all data from all subjects is already known and loaded.
+
+    .. image:: docs/source/images/withinsess.png
+        :alt: The schematic diagram of the WithinSession split
+        :align: center
+
+    Parameters
+    ----------
+    cross_val: cros-validation class, default=StratifiedKFold
+        Inner cross-validation strategy for splitting the sessions.
+    n_folds : int, default=5
+        Number of folds. Must be at least 2. If
+    random_state: int, RandomState instance or None, default=None
+        Controls the randomness of splits. Only used when `shuffle` is True.
+        Pass an int for reproducible output across multiple function calls.
+    shuffle : bool, default=True
+        Whether to shuffle each class's samples before splitting into batches.
+        Note that the samples within each split will not be shuffled.
+
+    """
+
+    def __init__(
+        self,
+        cross_val: BaseCrossValidator = StratifiedKFold,
```

Comment:
Last comment (see [doc](https://docs.python.org/3/library/typing.html#the-type-of-class-objects)), sorry I missed it before:
```suggestion
        cross_val: type[BaseCrossValidator] = StratifiedKFold,
```